api/spec/spec_handler.go

```diff
@@ -106,6 +106,7 @@ var (
 	haRegex                       = regexp.MustCompile(api.SpecHaLevel + "=([0-9]+),?")
 	cosRegex                      = regexp.MustCompile(api.SpecPriority + "=([A-Za-z]+),?")
 	sharedRegex                   = regexp.MustCompile(api.SpecShared + "=([A-Za-z]+),?")
+	sharedBlockRegex              = regexp.MustCompile(api.SpecSharedBlock + "=([A-Za-z]+),?")
```

Comment:
I would avoid defining anything here in spec handler since that results into a new storage class parameter which can be totally avoided by relying solely on PVC's accessMode & volumeMode fields.

Reply:
I carried forward existing framework. csi spec does indeed carry right recommendations. i will focus on getting this field properly initialized without special storageclass needs.